youtube_dl/extractor/xvideos.py

```diff
@@ -76,13 +76,24 @@ class XVideosIE(InfoExtractor):
     }, {
         'url': 'https://de.xvideos.com/video4588838/biker_takes_his_girl',
         'only_matching': True
+    }, {
+        'url': 'https://www.xvideos2.com/video4588838/biker_takes_his_girl',
+        'only_matching': True
+    }, {
+        'url': 'https://www.xvideos3.com/video4588838/biker_takes_his_girl',
+        'only_matching': True
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(
-            'https://www.xvideos.com/video%s/' % video_id, video_id)
+        HOSTS = ('xvideos', 'xvideos2', 'xvideos3')
+        self.report_download_webpage(video_id)
+        for host in HOSTS:
+            webpage = self._download_webpage(
+                'https://www.' + host + '.com/video%s/' % video_id, video_id, note=False, fatal=host == HOSTS[-1], errnote=host == HOSTS[-1])
+            if webpage:
```

Comment:
Also cache last accessible host and try it first.

Reply:
I dont think thats a good idea. In some videos the host xvideos contains more and better video formats than xvideos2 or 3. So it should always check xvideos first.

Reply:
Provide concrete examples.

Reply:
The first is with my VPN on (so xvideos) and second with it off (so xvideos2)

[xvideos3-add][~/ext/youtube-dl/youtube_dl]$ python __main__.py --list-format https://www.xvideos3.com/video29077757/                                         [XVideos] 29077757: Downloading webpage
[XVideos] 29077757: Downloading m3u8 information
[info] Available formats for 29077757:
format code  extension  resolution note
mp4-low      mp4        unknown
mp4-high     mp4        unknown
hls-250p     mp4        444x250     155k
hls-360p     mp4        640x360     423k
hls-480p     mp4        854x480     763k
hls-720p     mp4        1280x720   1327k  (best)
[xvideos3-add][~/ext/youtube-dl/youtube_dl]$ python __main__.py --list-format https://www.xvideos3.com/video29077757/
[XVideos] 29077757: Downloading webpage
[info] Available formats for 29077757:
format code  extension  resolution note
mp4-low      mp4        unknown
mp4-high     mp4        unknown    (best)

Reply:
Post `--dump-pages`.

Reply:
[vpn.txt](https://github.com/ytdl-org/youtube-dl/files/4575107/vpn.txt)
[novpn.txt](https://github.com/ytdl-org/youtube-dl/files/4575111/novpn.txt)

Reply:
Ok, then at least cache as class member field for subsequent extractions inside single run.

Reply:
will do but what to do about the failing test? Also shouldn't we warn the user that potentially lower quality video is being downloaded and ask to use vpn?

Reply:
1. Nothing, CI is running in unrestricted country.
2. This is not necessarily the case, for me all hosts return the same formats.